src/utils.cpp

```diff
@@ -144,6 +144,28 @@ std::string ToCIFS(std::string& UNCName)
   return CIFSname;
 }
 
+bool InsertUser(std::string& UNCName)
+{
+  if (g_szUser.empty())
+    return false;
+
+  std::string SMBPrefix;
+
+  if (!g_szPass.empty())
+    SMBPrefix = "smb://" + g_szUser + ":" + g_szPass + "@";
+  else
+     SMBPrefix = "smb://" + g_szUser + "@";
+
+  size_t found = UNCName.find("smb://");
+  if (found == 0)
```

Comment:
One last nitpick: the two lines above can be written as (you don't need the `found`variable, then):

`if (UNCName.find("smb://") == 0)`

Reply:
And this check can be done at the very beginning of the method, right after the user check. No need to assemble `SMBPrefix` in case `UNCName`does not start correctly.